rasa/core/training/interactive.py

```diff
@@ -590,9 +602,9 @@ async def _write_data_to_file(conversation_id: Text, endpoint: EndpointConfig):
     serialised_domain = await retrieve_domain(endpoint)
     domain = Domain.from_dict(serialised_domain)
 
-    _write_stories_to_file(story_path, events, domain)
-    _write_nlu_to_file(nlu_path, events)
-    _write_domain_to_file(domain_path, events, domain)
+    await _retry_on_error(_write_stories_to_file, story_path, events, domain)
```

Comment:
I think I'd prefer using lambda's here to avoid having to pass the args / kwargs to `_retry_on_error`. What do you think?

Reply:
I could use lambdas, but because I want to have access to the `export_path` in the error message, I would need to bind the `events` and `domain` vars in the lambda and then still pass `export_path` separately. It feels more convoluted to me. But I can change it if you like.

Reply:
Fair point 👍